rfcs/054-prismic-api/README.md

```diff
@@ -0,0 +1,110 @@
+- The new API service should be written in typescript, following patterns set by the [concepts API](../050-concepts-api/README.md) for filtering, pagination, error handling, etc.
+- We shouldn't create new IDs for exhibitions, events, stories, etc. There aren't any situations where we should need to merge content from multiple sources, so we should use the document IDs directly from prismic.
+- The elasticsearch index mapping should represent the contract between the pipeline and the API. The API shouldn't need to know anything about the structure of the data in prismic.
+
+## Proposed endpoints
+
+### `/articles`
+
+Filter for type eg stories, webcomics
+
+#### Different kinds of thing (are these 'labels'?)
+
+- story <https://wellcomecollection.org/articles/Y9vktREAACYA1v2U> at /articles
+- webcomic <https://wellcomecollection.org/articles/Y86rshEAACcAI4g4> at /articles
+- book <https://wellcomecollection.org/books/Y7QWrREAAHC43oH0> at /books
+- series <https://wellcomecollection.org/series/WfpEBCoAACrdd43y> composed of articles
+- in pictures <https://wellcomecollection.org/articles/Y-oAIhAAACAAix32> at /articles
+
+#### Filters
+
+- date
+- contributor
+- series
+- contentType
+
+#### Sort
+
+- date
+- relevance
+
+### `/exhibitions`
+
+#### Filter
+
+- date
+- status
+- location
+- contributor?
+- format (ie permanent, exhibition, installation)
+- season
+
+#### Sort
+
+- date
+- relevance
+
+### `/events`
+
+#### Filter
+
+- date
+- accessibility status (is it audio described, signed, etc)
+- location
+- format (eg discussion)
+- event-series
+- season
+- audience (eg schools)
+- contributor
+- isOnline?
+- availableOnline?
+
+#### Sort
+
+- date
+- relevance
+
+### `/books`
+
+?????????
+
+## Open questions
+
+- How do we represent functional content?
+- Should we include the full prismic API response at eg `/articles/{id}`?
+    - should we be able to list all of the docs of a given type at once? what should that url look like?
+- If we want to avoid maintaining multiple versions of the same code, should this work be done as an extension to the concepts API?
+- Works exist at [api.wellcomecollection.org/catalogue/v2/works](api.wellcomecollection.org/catalogue/v2/works), concepts at [/catalogue/v2/concepts](api.wellcomecollection.org/catalogue/v2/concepts), images at [/catalogue/v2/images](api.wellcomecollection.org/catalogue/v2/images).  
+Stories etc aren't part of the catalogue - should the URL be different?
+- what are the top level types? is it articles? stories? series?
+- where do books go? they appear on the stories landing page, so should they appear in stories search results? or are they at a separate endpoint?
+- what is a serial? what is a series?
+- is 'In pictures' the same as 'Image gallery'?
```

Comment:
@dannybirchall @davidpmccormick @gestchild one for you?

Reply:
There is also "Photo Story"

Reply:
where are these from?

Reply:
Labels!
See here for example, I think we might've gotten the wrong one [in search](https://wellcomecollection.org/search/stories?query=blue&page=2) where the card displays "Image Gallery" and then the [detailed page](https://wellcomecollection.org/articles/Ybi9_REAACAAcOrA) says "In Pictures"

Reply:
'In Pictures' is one of the story formats and has been added to the format field of the story in the example you've posted. Not sure where 'Image Gallery' is coming from yet, just looking.
N.B. there isn't an 'Image Gallery' format set up in Prismic.

Reply:
> There is also "Photo Story"

This is another format that can be applied to stories:

![Screenshot 2023-02-21 at 17 17 08](https://user-images.githubusercontent.com/6051896/220414977-62cf67b7-a6ed-47d7-b356-8e37f1418653.png)

Reply:
Looks like 'Image Gallery' is being displayed on search results, because [it is mapped to the id of the 'In pictures' story format](https://github.com/wellcomecollection/wellcomecollection.org/blob/6500012429db624e2f9fc9392bc6a1be41b6806e/common/data/content-format-ids.ts#L2)

And that is the [source of the label in the search result](https://github.com/wellcomecollection/wellcomecollection.org/blob/6500012429db624e2f9fc9392bc6a1be41b6806e/catalogue/webapp/services/prismic/fetch/index.ts#L17)

Don't know why that is hard coded and not coming from Prismic?

Reply:
So yes, 'In pictures' and 'Image gallery' are the same thing and the label on the search result needs fixing

Reply:
Thank you! Made a separate ticket for it: https://github.com/wellcomecollection/wellcomecollection.org/issues/9341